specifyweb/frontend/js_src/lib/components/AppResources/EditorWrapper.tsx

```diff
@@ -92,16 +92,20 @@ export function Wrapper({
     <NotFoundView container={false} />
   ) : (
     <AppResourceEditor
-      directory={directory}
+      directory={Array.isArray(directory) ? directory[0] : directory}
       initialData={initialData === false ? undefined : initialData}
       resource={record}
       onClone={(clonedResource, clone): void =>
         navigate(
           formatUrl(`${baseHref}/new/`, {
-            directoryKey: findAppResourceDirectoryKey(
-              resourcesTree,
-              directory.id
-            ),
+            directoryKey: Array.isArray(directory)
+              ? findAppResourceDirectoryKey(
+                  resourcesTree,
+                  directory.find(
+                    (dir) => dir.resource_uri === record.spAppResourceDir
+                  )?.id || -1
+                )
+              : findAppResourceDirectoryKey(resourcesTree, directory.id),
```

Comment:
Good job finding the issue.
But I would propose a much simpler solution

```suggestion
            directoryKey:
              directory.scope === 'global'
                ? globalResourcesKey
                : resourcesTreefindAppResourceDirectoryKey(
                    resourcesTree,
                    directory.id
                  ),

```

Where `globalResourcesKey` should be `export const globalResourcesKey = 'globalResources';`, and should be defined in tree.ts, and then also used in this place:

https://github.com/specify/specify7/blob/1858540f8c16eac398f39cca870abb55e346baa8/specifyweb/frontend/js_src/lib/components/AppResources/tree.ts#L40

by having a variable for this, any changes to the key would automatically update the places where it's used

and speaking of changes, feel free to rename the key value from `globalResources` to `globalResource` to better describe it's purpose (as there is only "one" global app resource)

Then, undo all other changes in this PR

Reply:
so much easier!!!